sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_client_async.py

```diff
@@ -55,68 +52,97 @@
+    :paramtype handle_max: int
+    :keyword on_attach: A callback function to be run on receipt of an ATTACH frame.
      The function must take 4 arguments: source, target, properties and error.
-    :type on_attach: func[~uamqp.address.Source, ~uamqp.address.Target, dict, ~uamqp.errors.AMQPConnectionError]
-    :param send_settle_mode: The mode by which to settle message send
+    :paramtype on_attach: func[~pyamqp.endpoint.Source, ~pyamqp.endpoint.Target, dict, ~pyamqp.error.AMQPConnectionError]
+    :keyword send_settle_mode: The mode by which to settle message send
      operations. If set to `Unsettled`, the client will wait for a confirmation
      from the service that the message was successfully sent. If set to 'Settled',
      the client will not wait for confirmation and assume success.
-    :type send_settle_mode: ~uamqp.constants.SenderSettleMode
-    :param receive_settle_mode: The mode by which to settle message receive
+    :paramtype send_settle_mode: ~pyamqp.constants.SenderSettleMode
+    :keyword receive_settle_mode: The mode by which to settle message receive
      operations. If set to `PeekLock`, the receiver will lock a message once received until
      the client accepts or rejects the message. If set to `ReceiveAndDelete`, the service
      will assume successful receipt of the message and clear it from the queue. The
      default is `PeekLock`.
-    :type receive_settle_mode: ~uamqp.constants.ReceiverSettleMode
-    :param encoding: The encoding to use for parameters supplied as strings.
-     Default is 'UTF-8'
-    :type encoding: str
+    :paramtype receive_settle_mode: ~pyamqp.constants.ReceiverSettleMode
+    :keyword desired_capabilities: The extension capabilities desired from the peer endpoint.
+     To create an desired_capabilities object, please do as follows:
+        - 1. Create an array of desired capability symbols: `capabilities_symbol_array = [types.AMQPSymbol(string)]`
+        - 2. Transform the array to AMQPValue object: `utils.data_factory(types.AMQPArray(capabilities_symbol_array))`
+    :paramtype desired_capabilities: List
+    :keyword max_message_size: The maximum allowed message size negotiated for the Link.	
+    :paramtype max_message_size: int	
+    :keyword link_properties: Metadata to be sent in the Link ATTACH frame.	
+    :paramtype link_properties: dict	
+    :keyword prefetch: The receiver Link credit that determines how many	
+     messages the Link will attempt to handle per connection iteration.	
+     The default is 300.	
+    :paramtype prefetch: int
+    :keyword transport_type: The type of transport protocol that will be used for communicating with
+     the service. Default is `TransportType.Amqp` in which case port 5671 is used.
+     If the port 5671 is unavailable/blocked in the network environment, `TransportType.AmqpOverWebsocket` could
+     be used instead which uses port 443 for communication.
+    :paramtype transport_type: ~pyamqp.constants.TransportType
```

Comment:
For EH, we actually pass in the TransportType defined in the sdk layer constants. I don't know if that TransportType falls under this one, but it's prob fine?

Reply:
I changed it b/c I figured we didnt want references to a specific service in pyamqp itself

Reply:
makes sense!